### PR TITLE
LPD-9361 New OOTB Object Type of Return

### PR DESCRIPTION
**Pull request by ghost**
https://liferay.atlassian.net/browse/LPD-9361

### CODE REVIEW

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Review by 4lejandrito — CHANGES_REQUESTED** [review]
Hey @CrescenzoRegaLF,

For the json file name [Object_Definition_Return.batch-engine-data.json](https://github.com/liferay-commerce/liferay-portal/pull/4675/files#diff-be80fb539189333ca3a9d5e9d1f401df08dcd39670c9688c90ab28a5f1edf956) I would suggest to use lowercase and hyphens as in:
* [headless-builder.batch-engine-data.json](https://github.com/liferay/liferay-portal/blob/e858dd6173797b0ac4467dd0a50bb28a26fc8b80/modules/apps/headless/headless-builder/headless-builder-impl/src/main/resources/com/liferay/headless/builder/internal/batch/headless-builder.batch-engine-data.json)
* [bookmarks.batch-engine-data.json](https://github.com/liferay/liferay-portal/blob/f48babb8eb0eedfc0e52a601a43607c5d162c13a/modules/apps/object/object-service/src/main/resources/com/liferay/object/internal/batch/bookmarks.batch-engine-data.json)

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by 4lejandrito** [comment]
Hey @CrescenzoRegaLF I think you may want to feature flag your changes. See [how we are doing it](https://github.com/liferay/liferay-portal/blob/e858dd6173797b0ac4467dd0a50bb28a26fc8b80/modules/apps/headless/headless-builder/headless-builder-impl/src/main/resources/com/liferay/headless/builder/internal/batch/headless-builder.batch-engine-data.json#L8). Your object definition will only be created once the FF is enabled.

**Comment by CrescenzoRegaLF** [comment]
Before adding a feature flag I must ask the product owner for confirmation

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:reevaluate:1397401_9406

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/returns.batch-engine-data.json [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+					"DBType": "String",
+					"businessType": "Picklist",
+					"externalReferenceCode": "RETURN-STATE",
+					"indexed": true,
+					"indexedAsKeyword": false,
+					"indexedLanguageId": "en_US",
+					"label": {
+						"en_US": "ReturnState"
+					},
+					"listTypeDefinitionExternalReferenceCode": "RETURN-STATE_PICKLIST",
```
Comment (4lejandrito) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/returns.batch-engine-data.json:36]: Hey @CrescenzoRegaLF it is better to prefix all the ERCs with `L_` not only the ones for object definitions (Brian told us that in a different PR).

**Review by 4lejandrito — CHANGES_REQUESTED** [review]

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
Hey @4lejandrito let me know if there are other changes or if it can be forwarded, thanks

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/returns.batch-engine-data.json [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+			"label": {
+				"en_US": "Return"
+			},
+			"modifiable": true,
+			"name": "Return",
+			"objectFields": [
+				{
+					"DBType": "String",
+					"businessType": "Picklist",
+					"externalReferenceCode": "L_RETURN-STATE",
```
Comment (4lejandrito) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/returns.batch-engine-data.json:29]: I would not mix `-` and `_` in the same ERC.

File: modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/returns.batch-engine-data.json [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+			"taskItemDelegateName": "DEFAULT",
+			"updateStrategy": "UPDATE"
+		},
+		"userId": 0,
+		"version": "v1.0"
+	},
+	"items": [
+		{
+			"enableCategorization": true,
+			"externalReferenceCode": "L_RETURN",
```
Comment (4lejandrito) [inline modules/apps/commerce/commerce-service/src/main/resources/com/liferay/commerce/internal/batch/returns.batch-engine-data.json:19]: I don't know what Brian will say, but Return sounds too generic. Maybe L_COMMERCE_ORDER_RETURN? This is just a suggestion, do what you want.

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Review by 4lejandrito — APPROVED** [review]

**Comment by CrescenzoRegaLF** [comment]
ci:test:sf

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:test:relevant

**Comment by CrescenzoRegaLF** [comment]
ci:forward:force

**Pull request closed without merging**